test/controllers/businessmatching/updateservice/add/WhatDoYouDoHereControllerSpec.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.businessmatching.updateservice.add
+
+import cats.data.OptionT
+import cats.implicits._
+import controllers.businessmatching.updateservice.UpdateServiceHelper
+import generators.businessmatching.BusinessMatchingGenerator
+import models.businessmatching._
+import models.flowmanagement.{AddServiceFlowModel, WhatDoYouDoHerePageId}
+import models.moneyservicebusiness.MoneyServiceBusinessTestData
+import org.mockito.Matchers._
+import org.mockito.Mockito._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.mock.MockitoSugar
+import play.api.i18n.Messages
+import play.api.test.Helpers._
+import services.businessmatching.BusinessMatchingService
+import uk.gov.hmrc.http.cache.client.CacheMap
+import utils.{AuthorisedFixture, DependencyMocks, GenericTestHelper}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class WhatDoYouDoHereControllerSpec extends GenericTestHelper with ScalaFutures with MockitoSugar with MoneyServiceBusinessTestData with BusinessMatchingGenerator {
+
+  sealed trait Fixture extends AuthorisedFixture with DependencyMocks {
+    self =>
+
+    val request = addToken(authRequest)
+
+    val mockBusinessMatchingService = mock[BusinessMatchingService]
+    val mockUpdateServiceHelper = mock[UpdateServiceHelper]
+
+    val controller = new WhatDoYouDoHereController(
+      authConnector = self.authConnector,
+      dataCacheConnector = mockCacheConnector,
+      statusService = mockStatusService,
+      businessMatchingService = mockBusinessMatchingService,
+      helper = mockUpdateServiceHelper,
+      router = createRouter[AddServiceFlowModel]
+    )
+
+    val cacheMapT = OptionT.some[Future, CacheMap](mockCacheMap)
+
+    when {
+      controller.dataCacheConnector.fetchAll(any(), any())
+    } thenReturn Future.successful(Some(mockCacheMap))
```

Comment:
This has been catered for already by bringing in `DependencyMocks` trait

Reply:
Done